include/deal.II/base/quadrature.h

```diff
@@ -221,6 +221,17 @@ class Quadrature : public Subscriptor
   template <class Archive>
   void serialize (Archive &ar, const unsigned int version);
 
+  /**
+   * Returns is_tensor_product_flag;
+   */
+  bool is_tensor_product() const;
+
+  /**
+   * In case the quadrature formula is a tensor product, this returns the one-dimensional
+   * basis object
+   */
+  Quadrature<1> get_tensor_basis() const;
```

Comment:
What does the function do if the quadrature formula is not a tensor product?

Since the function returns an object, not a reference, one may ask whether it is expensive to call. Would that be useful to point out in the documentation?